plugin/src/main/java/org/opensearch/ml/action/connector/GetConnectorTransportAction.java

```diff
@@ -77,6 +80,15 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLConn
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         Connector mlConnector = Connector.createConnector(parser);
                         mlConnector.removeCredential();
+                        if (!Objects.equals(tenantId, mlConnector.getTenantId())) {
+                            actionListener
+                                .onFailure(
+                                    new OpenSearchStatusException(
+                                        "You don't have permission to access this connector",
```

Comment:
When permission denied, maybe also adding the tenant id and the connector id and name to exception message, so it helps the users to get proper permission.

Reply:
This is a Get connector method where customer provided the connector id. So we shouldn't expose any unnecessary information from where customer can take advantage.